src/cache/cache_stats.hpp

```diff
@@ -0,0 +1,114 @@
+//--------------------------------------------------------------------------------------------------
+// Copyright (c) 2019 Alexey Sheplyakov
+//
+// This software is provided 'as-is', without any express or implied warranty. In no event will the
+// authors be held liable for any damages arising from the use of this software.
+//
+// Permission is granted to anyone to use this software for any purpose, including commercial
+// applications, and to alter it and redistribute it freely, subject to the following restrictions:
+//
+//  1. The origin of this software must not be misrepresented; you must not claim that you wrote
+//     the original software. If you use this software in a product, an acknowledgment in the
+//     product documentation would be appreciated but is not required.
+//
+//  2. Altered source versions must be plainly marked as such, and must not be misrepresented as
+//     being the original software.
+//
+//  3. This notice may not be removed or altered from any source distribution.
+//--------------------------------------------------------------------------------------------------
+
+#ifndef BUILDCACHE_CACHE_STATS_HPP_
+#define BUILDCACHE_CACHE_STATS_HPP_
+#include <iosfwd>
+#include <string>
+
+struct cJSON;
+
+namespace bcache {
+
+struct cache_stats_t {
+  int local_miss_count{0};
+  int local_hit_count{0};
+  int remote_hit_count{0};
+  int remote_miss_count{0};
+  bool reliable{true};
+
+  bool fromFile(const std::string& path) noexcept;
+  bool fromJson(cJSON const* obj) noexcept;
+  bool toJson(cJSON* obj) const noexcept;
+  bool toFile(const std::string& path) const noexcept;
```

Comment:
Please use snake_case for all method names (i.e. `from_file`, `from_json`, `to_json` and `to_file`).

Reply:
done